### ISSUE TITLE
We should be able to handle a panic (trigger error reporting) if it comes from `getDMMF`

### ISSUE THREAD

**Issue by Jolg42** [prisma/prisma#10206]
This was only confirmed using the Node-API engine when using getDMMF (to be confirmed if binary engine is also affected)

As of today a panic happening in `getDmmfNodeAPI` looks like this 

(same ~ output for `validate`, `migrate dev` or `db push`)
```
npx prisma@latest db push
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "repropanicvalidate", schema "public" at "localhost:5432"

PostgreSQL database repropanicvalidate created at localhost:5432
Error: db error: ERROR: syntax error at or near ")"
   0: sql_migration_connector::sql_database_step_applier::apply_migration
             at migration-engine/connectors/sql-migration-connector/src/sql_database_step_applier.rs:11
   1: migration_core::api::SchemaPush
             at migration-engine/core/src/api.rs:184 
```

If we fix it it will look like this (here DEBUG is enabled with extra logging)
<img width="671" alt="Screen Shot 2021-11-11 at 09 51 03" src="https://user-images.githubusercontent.com/1328733/141268204-493438b4-c712-4dd6-8e48-c7c2f41812bb.png">

I created a small PR to demonstrate what we need to do:
https://github.com/prisma/prisma/pull/10205/files

How to reproduce?
See https://github.com/prisma/prisma/issues/10118#issuecomment-965641590

In a nutshell
- create this schema (do not auto format)
```prisma
datasource db {
    provider = "postgresql"
    url      = "postgres://prisma:prisma@localhost:5432/repropanicvalidate"
}

generator client {
    provider = "prisma-client-js"
}

model User {
  id         String    @id @default(cuid()) @db.Char(30)
  referralId String    @unique @db.Char(30)
  referral   Referral? @relation("UserToReferral", fields: [referralId], references: [id])
}
model Referral {
  id   String @id @default(cuid()) @db.Char(30)
  user User   @relation("UserToReferral")
}
```
- run `prisma validate`

**Comment by janpio** [comment]
Replaced by https://github.com/prisma/prisma/issues/12495
